src/curl.c

```diff
@@ -118,6 +118,20 @@ int swupd_curl_init(void)
 	return 0;
 }
 
+CURLcode curl_set_optional_client_cert(CURL* curl, char* url) {
+	CURLcode curl_ret = CURLE_OK;
+	if (strncmp(url, "https://", 8) == 0) {
+		curl_ret = curl_easy_setopt(curl, CURLOPT_SSLCERT, SSL_CLIENT_CERT);
+		if (curl_ret == CURLE_SSL_CERTPROBLEM) {
```

Comment:
Is that also the return when the file does not exist?

Reply:
Looks like it isn't.  Fixing.

Reply:
This is more difficult to fix than expected.  Performing a `curl_easy_setopt` returns 0 even when the client SSL certificate doesn't exist.  Only at runtime with `curl_easy_perform` does the proper error code I'm looking for come up (58).  This is still a TOCTOU issue.  After speaking with @matthewrsj, we decided to go back to the original checking implementation.  The idea is that a SSL client cert is an opt-in construct.  If it's there at the time of check, but disappears later, or is not there but then appears later, is on the user to manage correctly.

Reply:
```
#include <curl/curl.h>
#include <errno.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

CURLcode curl_set_optional_client_cert(CURL* curl, char* url) {
	CURLcode curl_ret = CURLE_OK;
	if (strncmp(url, "https://", 8) == 0) {
		curl_ret = curl_easy_setopt(curl, CURLOPT_SSLCERTTYPE, "PEM");
		curl_ret = curl_easy_setopt(curl, CURLOPT_SSLCERT, "/tmp/lol.pem");
		fprintf(stderr, "Curl error: (%d) %s\n", curl_ret, curl_easy_strerror(curl_ret));
		if (curl_ret == CURLE_SSL_CERTPROBLEM) {
			curl_ret = CURLE_OK;
		}
	}
	return curl_ret;
}

int main() {
	curl_global_init(CURL_GLOBAL_ALL);
	CURL* curl = curl_easy_init();
	char* url = "https://download.clearlinux.org/update/";
	curl_set_optional_client_cert(curl, url);
}
```

Save as test.c and then run `gcc test.c -lcurl -o test`.  This is the program I wrote to test this behavior.